public/app/features/dashboard-scene/scene/NavToolbarActions.tsx

```diff
@@ -138,12 +138,17 @@ export function ToolbarActions({ dashboard }: Props) {
   toolbarActions.push({
     group: 'icon-actions',
     condition: meta.isSnapshot && !meta.dashboardNotFound && !isEditing,
-    render: () => (
-      <GoToSnapshotOriginButton
-        key="go-to-snapshot-origin"
-        originalURL={dashboard.getInitialSaveModel()?.snapshot?.originalUrl ?? ''}
-      />
-    ),
+    render: () => {
+      const saveModel = dashboard.getInitialSaveModel();
+
+      if (saveModel && isV2Dashboard(saveModel)) {
+        throw new Error('v2 schema not implemented');
+      }
+
+      return (
+        <GoToSnapshotOriginButton key="go-to-snapshot-origin" originalURL={saveModel?.snapshot?.originalUrl ?? ''} />
+      );
```

Comment:
nitpick: we could also add getSnapshotSourceUrl to serializer interface (optional and only if implemented show the button)